tests/tests.cairo

```diff
@@ -1,139 +1,154 @@
-use raito::validation::bits_to_target;
-use raito::validation::target_to_bits;
+use super::blocks::{block_0::block_0, block_170::block_170};
+use raito::state::{ChainState, UtreexoState};
+use raito::validation::{BlockValidatorImpl};
+
+#[test]
+fn test_block0() {
+    let block = block_0();
+    let prev_chain_state = ChainState {
+        block_height: 0, // cannot set to -1 or u32_max
```

Comment:
We need a special rule for genesis validation then.
Tbh I'm not sure we need to validate the genesis block in general, but for now let's just handle this special case in the validate_and_apply. Alternatively we can use `i64` or `Option<u32>` though it's a bit more friction